### ISSUE TITLE
PB626: Screen refresh issue

### ISSUE THREAD

**Issue by sonix-github** [koreader/koreader#4201]
* KOReader version: v2015.11-1840-gababbd7_2018-09-06
* Device: PocketBook 626 with software E626.5.14.1227

#### Issue
UI screen is not updated properly

#### Steps to reproduce
1. Open any ebook e.g. PDF file
2. Tap on UI top and tap on file browser icon
3. File browser icon stays selected (inverted) and file browser is NOT visible on screen
-if not occurs the issue repeat steps 1-3
4. Tap anywhere on screen or press up/down hardware button and screen immediately updates but it processes action as when it is visible. So when I push up or down hardware button it changes to previous or next page in file browser.

Similar happens sometimes by opening e-book. Open file message stays visible and no opened e-book is visible. Again if I press up/down button it immediately updates the screen but also moves to previous or next page.

And screen update issue is also present in file browser after new e-book is copied from PC to PocketBook. If I browse through folder and see the file list where the new book is, the metadata (book author) is not automatically updated on screen. And if I press up/down button it immediately updates the metadata on screen just a fraction of second before it moves to previous or next page.

It was NOT present in version v2015.11-1523-g78284e1 from 12.2.2018

##### `crash.log` (if applicable)
[crash.log](https://github.com/koreader/koreader/files/2362134/crash.log)

**Comment by NiLuJe** [comment]
Hard to say much more without being familiar with PB devices.

You can try to enable debug logging, it might shed some more light on the specific cases when things go wrong...

**Comment by sonix-github** [comment]
No, there was no FW update between KOReader updates. I will try to enable debug logging to see if there is something which can help clarify this issue.

**Comment by sonix-github** [comment]
I enabled debug logging (with -d flag) and reproduced the issue. It occurred (file browser icon stayed inverted and screen was not updated with book content automatically only after pressing down button) on the 4-th e-book opening.

Attached is crash log file.

[crash_20180907_debug_mode_enabled.log](https://github.com/koreader/koreader/files/2362347/crash_20180907_debug_mode_enabled.log)

**Comment by NiLuJe** [comment]
@sonix-github : Did you switch orientation for that 4th PDF?

**Comment by sonix-github** [comment]
@NiLuJe : No I did not changed orientation during this screen update issue reproduction. I only opened files from file browser.

**Comment by NiLuJe** [comment]
Then there's definitely an orientation issue on your device, because the orientation appears to be all over the place.

**Comment by sonix-github** [comment]
@NiLuJe: If have you read **all three** situations (described in my initial contribution) when this issue happens then I do not understand what brings you to conclusion about orientation issue in this case. Something in KOReader (after version  v2015.11-1523-g78284e1 from 12.2.2018) causes NOT to fire refresh event in some situations.

**Comment by NiLuJe** [comment]
```
09/07/18-21:59:28 INFO  opening file /mnt/ext1/Moje eBooks/Technické/Katalógové listy/2SK2885.pdf
ffi.load: libs/liblept.so.4
ffi.load: libs/libk2pdfopt.so.2
09/07/18-21:59:29 DEBUG data is read from  /mnt/ext1/SDR Metadata/2SK2885.sdr/metadata.pdf.lua
09/07/18-21:59:29 DEBUG ImageWidget: _render'ing resources/icons/appbar.book.open.png
09/07/18-21:59:29 DEBUG renderImageData: using MuPDF
09/07/18-21:59:29 DEBUG Mupdf.renderImage true cdata<struct BlitBufferRGB24>: 0x40230318
09/07/18-21:59:29 DEBUG cache image|resources/icons/appbar.book.open.png||
09/07/18-21:59:29 INFO  RD loaded plugin batterystat at plugins/batterystat.koplugin
09/07/18-21:59:29 INFO  RD loaded plugin coverbrowser at plugins/coverbrowser.koplugin
09/07/18-21:59:29 INFO  RD loaded plugin perceptionexpander at plugins/perceptionexpander.koplugin
09/07/18-21:59:29 INFO  RD loaded plugin readtimer at plugins/readtimer.koplugin
09/07/18-21:59:29 INFO  RD loaded plugin statistics at plugins/statistics.koplugin
09/07/18-21:59:29 INFO  RD loaded plugin systemstat at plugins/systemstat.koplugin
09/07/18-21:59:29 INFO  RD loaded plugin terminal at plugins/terminal.koplugin
09/07/18-21:59:29 INFO  RD loaded plugin texteditor at plugins/texteditor.koplugin
09/07/18-21:59:29 INFO  RD loaded plugin timesync at plugins/timesync.koplugin
09/07/18-21:59:29 DEBUG update_mode: Update refresh mode partial to full
09/07/18-21:59:29 DEBUG _refresh: Enqueued full update for region 0 0 1024 1024
09/07/18-21:59:29 DEBUG setDirty full from widget ReaderUI w/ NO region
09/07/18-21:59:29 DEBUG _refresh: Enqueued full update for region 0 0 1024 1024
09/07/18-21:59:29 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-21:59:29 DEBUG _refresh: Enqueued full update for region 0 0 1024 1024
09/07/18-21:59:29 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-21:59:29 DEBUG init scroll page states
09/07/18-21:59:29 INFO  setting zoom mode to contentwidth
source page (0,0) - (497,711)
source page bgcolor 192
09/07/18-21:59:29 DEBUG _refresh: Enqueued full update for region 0 0 1024 1024
09/07/18-21:59:29 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-21:59:29 DEBUG _refresh: Enqueued full update for region 0 0 1024 1024
09/07/18-21:59:29 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-21:59:29 DEBUG init scroll page states
09/07/18-21:59:29 DEBUG get page position
09/07/18-21:59:29 DEBUG _refresh: Enqueued full update for region 0 0 1024 1024
09/07/18-21:59:29 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-21:59:29 DEBUG init scroll page states
09/07/18-21:59:29 DEBUG get page position
09/07/18-21:59:29 DEBUG _refresh: Enqueued full update for region 0 0 1024 1024
09/07/18-21:59:29 DEBUG setDirty ui from widget ReaderUI w/ NO region
09/07/18-21:59:29 DEBUG _refresh: Enqueued full update for region 0 0 1024 1024
09/07/18-21:59:29 DEBUG setDirty ui from widget ReaderUI w/ NO region
09/07/18-21:59:29 DEBUG show widget ReaderUI
09/07/18-21:59:29 DEBUG setDirty nil from widget ReaderUI w/ NO region
09/07/18-21:59:29 DEBUG close filemanager
09/07/18-21:59:29 DEBUG close widget
09/07/18-21:59:29 DEBUG CoverMenu:onCloseWidget: terminating jobs if needed
09/07/18-21:59:29 DEBUG terminating 0 subprocesses
09/07/18-21:59:29 DEBUG TextBoxWidget:free called
09/07/18-21:59:29 DEBUG TextBoxWidget:free called
09/07/18-21:59:29 DEBUG TextBoxWidget:free called
09/07/18-21:59:29 DEBUG TextBoxWidget:free called
09/07/18-21:59:29 DEBUG TextBoxWidget:free called
09/07/18-21:59:29 DEBUG TextBoxWidget:free called
09/07/18-21:59:29 DEBUG TextBoxWidget:free called
09/07/18-21:59:29 DEBUG TextBoxWidget:free called
09/07/18-21:59:29 DEBUG TextBoxWidget:free called
09/07/18-21:59:29 DEBUG TextBoxWidget:free called
09/07/18-21:59:29 DEBUG TextBoxWidget:free called
09/07/18-21:59:29 DEBUG TextBoxWidget:free called
09/07/18-21:59:29 DEBUG _refresh: Enqueued full update for region 0 0 1024 1024
09/07/18-21:59:29 DEBUG setDirty partial from widget nil w/ NO region
09/07/18-21:59:29 DEBUG setDirty nil from widget ReaderUI w/ NO region
09/07/18-21:59:29 DEBUG painting widget: ReaderUI
09/07/18-21:59:30 DEBUG _refresh: Enqueued full update for region 0 0 1024 1024
09/07/18-21:59:30 DEBUG refresh: full -1 -1 1026 1026
09/07/18-21:59:30 DEBUG refresh: wait for completion of (previous) marker 44 with collision_test 0
09/07/18-21:59:30 DEBUG refresh: wait for completion of marker 45 with collision_test 0
```

First book, 1024x1024 makes no sense, but works by some miracle of nature.

```
09/07/18-21:59:47 INFO  opening file /mnt/ext1/Moje eBooks/Technické/Katalógové listy/2SK2876.pdf
09/07/18-21:59:47 DEBUG data is read from  /mnt/ext1/SDR Metadata/2SK2876.sdr/metadata.pdf.lua
09/07/18-21:59:47 DEBUG ImageWidget: _render'ing resources/icons/appbar.book.open.png
09/07/18-21:59:47 INFO  RD loaded plugin batterystat at plugins/batterystat.koplugin
09/07/18-21:59:47 INFO  RD loaded plugin coverbrowser at plugins/coverbrowser.koplugin
09/07/18-21:59:47 INFO  RD loaded plugin perceptionexpander at plugins/perceptionexpander.koplugin
09/07/18-21:59:47 INFO  RD loaded plugin readtimer at plugins/readtimer.koplugin
09/07/18-21:59:47 INFO  RD loaded plugin statistics at plugins/statistics.koplugin
09/07/18-21:59:47 INFO  RD loaded plugin systemstat at plugins/systemstat.koplugin
09/07/18-21:59:47 INFO  RD loaded plugin terminal at plugins/terminal.koplugin
09/07/18-21:59:47 INFO  RD loaded plugin texteditor at plugins/texteditor.koplugin
09/07/18-21:59:47 INFO  RD loaded plugin timesync at plugins/timesync.koplugin
09/07/18-21:59:47 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-21:59:47 DEBUG setDirty full from widget ReaderUI w/ NO region
09/07/18-21:59:47 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-21:59:47 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-21:59:47 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-21:59:47 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-21:59:47 DEBUG init scroll page states
09/07/18-21:59:47 INFO  setting zoom mode to contentwidth
source page (0,0) - (594,841)
source page bgcolor 192
09/07/18-21:59:47 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-21:59:47 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-21:59:48 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-21:59:48 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-21:59:48 DEBUG init scroll page states
09/07/18-21:59:48 DEBUG get page position
09/07/18-21:59:48 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-21:59:48 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-21:59:48 DEBUG init scroll page states
09/07/18-21:59:48 DEBUG get page position
09/07/18-21:59:48 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-21:59:48 DEBUG setDirty ui from widget ReaderUI w/ NO region
09/07/18-21:59:48 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-21:59:48 DEBUG setDirty ui from widget ReaderUI w/ NO region
09/07/18-21:59:48 DEBUG show widget ReaderUI
09/07/18-21:59:48 DEBUG setDirty nil from widget ReaderUI w/ NO region
09/07/18-21:59:48 DEBUG painting widget: ReaderUI
09/07/18-21:59:48 DEBUG loading tile cache from ./cache/2227d224db4980f35b5f9237b280791d {
    ["pageno"] = 1,
    ["excerpt"] = {
        ["y"] = 0,
        ["x"] = 0,
        ["h"] = 1627,
        ["w"] = 1150
    },
    ["size"] = 1871114,
    ["bb"] = 
}
09/07/18-21:59:48 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-21:59:48 DEBUG refresh: full -1 -1 1026 760
09/07/18-21:59:48 DEBUG refresh: wait for completion of (previous) marker 49 with collision_test 0
09/07/18-21:59:48 DEBUG refresh: wait for completion of marker 50 with collision_test 0
```

Second book, 1024x758 makes sense... in landscape mode.
Same for the third book:

```
09/07/18-22:00:01 INFO  opening file /mnt/ext1/Moje eBooks/Technické/Katalógové listy/2SK2885.pdf
09/07/18-22:00:02 DEBUG data is read from  /mnt/ext1/SDR Metadata/2SK2885.sdr/metadata.pdf.lua
09/07/18-22:00:02 DEBUG ImageWidget: _render'ing resources/icons/appbar.book.open.png
09/07/18-22:00:02 INFO  RD loaded plugin batterystat at plugins/batterystat.koplugin
09/07/18-22:00:02 INFO  RD loaded plugin coverbrowser at plugins/coverbrowser.koplugin
09/07/18-22:00:02 INFO  RD loaded plugin perceptionexpander at plugins/perceptionexpander.koplugin
09/07/18-22:00:02 INFO  RD loaded plugin readtimer at plugins/readtimer.koplugin
09/07/18-22:00:02 INFO  RD loaded plugin statistics at plugins/statistics.koplugin
09/07/18-22:00:02 INFO  RD loaded plugin systemstat at plugins/systemstat.koplugin
09/07/18-22:00:02 INFO  RD loaded plugin terminal at plugins/terminal.koplugin
09/07/18-22:00:02 INFO  RD loaded plugin texteditor at plugins/texteditor.koplugin
09/07/18-22:00:02 INFO  RD loaded plugin timesync at plugins/timesync.koplugin
09/07/18-22:00:02 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-22:00:02 DEBUG setDirty full from widget ReaderUI w/ NO region
09/07/18-22:00:02 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-22:00:02 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:02 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-22:00:02 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:02 DEBUG init scroll page states
09/07/18-22:00:02 INFO  setting zoom mode to contentwidth
09/07/18-22:00:02 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-22:00:02 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:02 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-22:00:02 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:02 DEBUG init scroll page states
09/07/18-22:00:02 DEBUG get page position
09/07/18-22:00:02 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-22:00:02 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:02 DEBUG init scroll page states
09/07/18-22:00:02 DEBUG get page position
09/07/18-22:00:02 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-22:00:02 DEBUG setDirty ui from widget ReaderUI w/ NO region
09/07/18-22:00:02 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-22:00:02 DEBUG setDirty ui from widget ReaderUI w/ NO region
09/07/18-22:00:02 DEBUG show widget ReaderUI
09/07/18-22:00:02 DEBUG setDirty nil from widget ReaderUI w/ NO region
09/07/18-22:00:02 DEBUG painting widget: ReaderUI
09/07/18-22:00:02 DEBUG _refresh: Enqueued full update for region 0 0 1024 758
09/07/18-22:00:02 DEBUG refresh: full -1 -1 1026 760
09/07/18-22:00:02 DEBUG refresh: wait for completion of (previous) marker 54 with collision_test 0
09/07/18-22:00:02 DEBUG refresh: wait for completion of marker 55 with collision_test 0
```

But the fourth book...
```
09/07/18-22:00:14 INFO  opening file /mnt/ext1/Moje eBooks/Technické/Katalógové listy/2SK2750.pdf
09/07/18-22:00:14 DEBUG ImageWidget: _render'ing resources/icons/appbar.book.open.png
09/07/18-22:00:14 INFO  RD loaded plugin batterystat at plugins/batterystat.koplugin
09/07/18-22:00:14 INFO  RD loaded plugin coverbrowser at plugins/coverbrowser.koplugin
09/07/18-22:00:14 INFO  RD loaded plugin perceptionexpander at plugins/perceptionexpander.koplugin
09/07/18-22:00:14 INFO  RD loaded plugin readtimer at plugins/readtimer.koplugin
09/07/18-22:00:14 INFO  RD loaded plugin statistics at plugins/statistics.koplugin
09/07/18-22:00:14 INFO  RD loaded plugin systemstat at plugins/systemstat.koplugin
09/07/18-22:00:14 INFO  RD loaded plugin terminal at plugins/terminal.koplugin
09/07/18-22:00:14 INFO  RD loaded plugin texteditor at plugins/texteditor.koplugin
09/07/18-22:00:14 INFO  RD loaded plugin timesync at plugins/timesync.koplugin
09/07/18-22:00:14 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:14 DEBUG setDirty full from widget ReaderUI w/ NO region
09/07/18-22:00:14 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:14 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG init scroll page states
09/07/18-22:00:15 INFO  setting zoom mode to page
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG init scroll page states
09/07/18-22:00:15 DEBUG get page position
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG init scroll page states
09/07/18-22:00:15 DEBUG get page position
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty partial from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty ui from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:15 DEBUG setDirty ui from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG show widget ReaderUI
09/07/18-22:00:15 DEBUG setDirty nil from widget ReaderUI w/ NO region
09/07/18-22:00:15 DEBUG painting widget: ReaderUI
09/07/18-22:00:16 DEBUG _refresh: Enqueued full update for region 0 0 758 1024
09/07/18-22:00:16 DEBUG refresh: full -1 -1 760 1026
09/07/18-22:00:16 DEBUG refresh: wait for completion of (previous) marker 59 with collision_test 0
09/07/18-22:00:16 DEBUG refresh: wait for completion of marker 60 with collision_test 0
```

758x1024 makes sense, but it's Portrait.

**Comment by NiLuJe** [comment]
Note that I added some more debug logging around this in today's snapshots, but some truly weird shit is happening here, and I have no clue why (again, PB >_<").

**Comment by NiLuJe** [comment]
Also, if the ioctl were actually truly failing, that'd be logged, so the PB fb/driver is once again blatantly lying to our face.

**Comment by sonix-github** [comment]
Today I copied last KOReader version (v2015.11-1844-g51382d5_2018-09-08) and made few tests.

I opened 4 times the same PDF file and on the 4th tap on file browser icon it did not refresh the screen and the icon stayed inverted. Then I pressed hardware button "down" and immediately (just before moving to next page in file browser list) the screen has refreshed.

[crash_20180910_v2015.11-1844-g51382d5_2018-09-08.log](https://github.com/koreader/koreader/files/2367861/crash_20180910_v2015.11-1844-g51382d5_2018-09-08.log)

**Comment by NiLuJe** [comment]
Re-using the same book was a good idea ;).

No strange rotation behavior this time, so there's that at least...

Unfortunately, I can't see anything out of the ordinary, or anything that might make the 4th opening different. I see the refresh, and it reportedly succeeded, so I don't really know what might be going wrong...

**Comment by NiLuJe** [comment]
The only thing I can think of is a weird PB-specific aversion to A2 updates, because that vaaaaaguely rings a bell, so see if tweaking [this line](https://github.com/koreader/koreader-base/blob/5da954e47e64c163e96d6202b5efa47cd5c88c8c/ffi/framebuffer_mxcfb.lua#L515) to `self.waveform_fast = C.WAVEFORM_MODE_GC16` helps?

**Comment by poire-z** [comment]
Dunno anything about PB, but could be related to #3561 ?

**Comment by sonix-github** [comment]
@NiLuJe : Changing framebuffer_mxcfb.lua at line to self.waveform_fast = C.WAVEFORM_MODE_GC16 did not change and the issue persists.

**Comment by sonix-github** [comment]
> 
> 
> Dunno anything about PB, but could be related to #3561 ?

I have not observed similar behavior as in #3561. My PB626 updates time on status bar only by page turning or another menu action.

**Comment by poire-z** [comment]
May be not the specific symptoms like time update (it says _But you enabled 'Auto refresh time'?_ to have it), but could it be it suspends itself after a few seconds of inactivity?

> I opened 4 times the same PDF file and on the 4th tap on file browser icon it did not refresh the screen

could be explained if you were a bit slower to tap on the 4th opening :)

**Comment by sonix-github** [comment]
@poire-z : I overlooked information about enabling auto refresh time in KOReader settings. Now I enabled it but it did not refresh the time in status bar. Then I changed in PB menu settings standby mode from 10 minutes to off. After that change it automatically refreshed time only for one minute and then it refreshes only with page turning or other menu actions. How can I help to debug it?

**Comment by mk-f** [comment]
@sonix-github see https://github.com/koreader/koreader/pull/3540#issuecomment-429649687 and [this](https://github.com/koreader/koreader-base/blob/3488223893d76a3ec045ed92003cdf4322474c43/input/input-pocketbook.h#L83) .TL;DR PB auto-suspends the reader after ~2 seconds. I did some work which disables this for the first 60 seconds, hoping that this would be long enough to open most books. After that, and if not disabled by a yet-to-be-written plugin, the suspend is re-enabled to not drain battery. All kinds of UI problems are related to this, because the suspend might kick in while a page is redrawn.

**Comment by sonix-github** [comment]
@mk-f : Thanks for clarification. Now it makes more sense :-) I did not see this "behavior" in the past in Koreader. If there is some progress in plugin development I can test it at my PB626.

**Comment by pazos** [comment]
Duplicate of #3561
